prow/githuboauth/githuboauth.go

```diff
@@ -54,13 +54,44 @@ type GitHubClientGetter interface {
 
 // OAuthClient is an interface for a GitHub OAuth client.
 type OAuthClient interface {
+	WithFinalRedirectURL(url string) OAuthClient
 	// Exchanges code from GitHub OAuth redirect for user access token.
 	Exchange(ctx context.Context, code string, opts ...oauth2.AuthCodeOption) (*oauth2.Token, error)
 	// Returns a URL to GitHub's OAuth 2.0 consent page. The state is a token to protect the user
 	// from an XSRF attack.
 	AuthCodeURL(state string, opts ...oauth2.AuthCodeOption) string
 }
 
+type client struct {
+	config *oauth2.Config
+}
+
+func NewClient(config *oauth2.Config) client {
+	return client{
+		config: config,
+	}
+}
+
+func (cli client) WithFinalRedirectURL(url string) OAuthClient {
+	return NewClient(
+		&oauth2.Config{
+			ClientID:     cli.config.ClientID,
+			ClientSecret: cli.config.ClientSecret,
+			RedirectURL:  cli.config.RedirectURL + url,
```

Comment:
Is string concatenation always safe here or can we create an invalid URL by doing so?

Reply:
(what if someone doesn't register a trailing slash on the redirect URL?)